.github/workflows/pr-check-update-e2e.yaml

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: pr-check-update-e2e
+
+on: 
+  pull_request:
+    paths:
+      - '**'
+      - 'pnpm-lock.yaml'
+    types: [labeled, synchronize, opened, reopened]
+
+jobs:
+  detect_pnpm_changes:
+    name: Detect pnpm lock file changes
+    runs-on: ubuntu-latest
+    outputs:
+      pnpm_lock_changed: ${{ steps.pnpm_changed.outputs.PNPM_LOCK_CHANGED }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Evaluate changes in files
+        id: pnpm_changed
+        run: |
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+          git diff --name-only origin/${{ github.event.pull_request.base.ref }} HEAD > changes.txt
+          if grep -q 'pnpm-lock.yaml' changes.txt; then
+            echo "PNPM_LOCK_CHANGED=true" >> $GITHUB_OUTPUT
+          else
+            echo "PNPM_LOCK_CHANGED=false" >> $GITHUB_OUTPUT
+          fi
+
+  run-update-e2e-test:
+    name: win update e2e tests
+    needs: detect_pnpm_changes
+    if: contains(github.event.pull_request.labels.*.name, 'area/update') || needs.detect_pnpm_changes.outputs.pnpm_lock_changed == 'true'
+    runs-on: ${{ matrix.os }}
```

Comment:
why do we have a matrix if title and os are hardcoded to windows ?

Reply:
if matrix is used, name should use matrix.os and not win

Reply:
I was a step ahead, will keep it for windows for now.